boa/src/object/internal_methods.rs

```diff
@@ -336,7 +336,7 @@ impl Object {
 
     /// Helper function for property insertion.
     #[inline]
-    pub(crate) fn insert_property<K>(&mut self, key: K, property: Property) -> Option<Property>
+    pub(crate) fn insert<K>(&mut self, key: K, property: Property) -> Option<Property>
```

Comment:
`Object::insert`  inserts a `Property`, but `Object::insert_property`, inserts `V: Into<Value>`.
Seems pretty confusing to me, should switch them up, no?

Reply:
> `Object::insert` inserts a `Property`, but `Object::insert_property`, inserts `V: Into<Value>`.
> Seems pretty confusing to me, should switch them up, no?

Its is kind of confuzing but `inset` does not insert a property it inserts a Property Descriptor (this is why `Property` should be called `PropertyDescriptor` which is the way object as stored as a `HashMap<PropertyKey, PropertyDescriptor>`), and `insert_property` inserts a property like `x: "foo"`